gazelle_haskell_modules/rule_generation.go

```diff
@@ -393,6 +431,26 @@ func getSources(expr build.Expr) (map[string]bool, error) {
 	return sourceMap, nil
 }
 
+func getSourcesRecursivelyFromDirs(pkgRoot string, dirs []string) (map[string]bool, error) {
+	sourceMap := make(map[string]bool)
+	for _, dir := range dirs {
+		err := filepath.WalkDir(path.Join(pkgRoot, dir), func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				log.Fatal(err)
+			}
+			// TODO what about hsc and -boot files?
```

Comment:
Not sure what the proper thing to do here is. Any tips?

Reply:
boot files could be added as part of #16.

Maybe this could function could discover hsc files too.

Reply:
What about `hsc`? Do they need some special treatment compared to "normal" haskell modules for `haskell_module`?

Reply:
That's a good question. I don't know. I just found that no test in `rules_haskell` is trying `haskell_module` on `hsc` files.

Reply:
They seem to be tested in https://github.com/tweag/rules_haskell/tree/master/tests/hsc so I'll include them.

Reply:
Err...
```
ERROR: /home/googleson78/git/gazelle_haskell_modules/example/package-c/BUILD.bazel:191:15: in src attribute of _haskell_module rule //package-c:package-c-hsc.Hsc: source file '//package-c:hsc/Hsc.hsc' is misplaced here (expected .hs, .lhs, .hs-boot or .lhs-boot). Since this rule was created by the macro 'haskell_module', the error might have been caused by the macro implementation
```

Reply:
I'll revert it and open an issue to track hsc support.